optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/pages/employeeroster/EmployeeRosterPage.java

```diff
@@ -0,0 +1,191 @@
+    private EmployeeAvailabilityBlobPopoverContent availabilityBlobPopoverContent;
+
+    @Inject
+    private EmployeeRosterViewportFactory employeeRosterViewportFactory;
+
+    @Inject
+    private TenantStore tenantStore;
+
+    @Inject
+    private LoadingSpinner loadingSpinner;
+
+    private EmployeeRosterViewport viewport;
+    private EmployeeRosterView currentEmployeeRosterView;
+    //private Pagination spotsPagination = Pagination.of(0, 10);
+
+    @PostConstruct
+    public void init() {
+        availabilityBlobPopover.init(this, availabilityBlobPopoverContent);
+    }
+
+    public BlobPopover getBlobPopover() {
+        return availabilityBlobPopover;
+    }
+
+    @Override
+    public Promise<Void> beforeOpen() {
+        return refreshWithLoadingSpinner();
+    }
+
+    public void onTenantChanged(@Observes final TenantStore.TenantChange tenant) {
+        refreshWithLoadingSpinner();
+    }
+
+    private Promise<Void> refreshWithoutLoadingSpinner() {
+        return fetchEmployeeRosterView().then(employeeRosterView -> {
+            viewport = employeeRosterViewportFactory.getViewport(employeeRosterView);
+            viewportView.setViewport(viewport);
+            return resolve();
+
+        });
+    }
+
+    private Promise<Void> refreshWithLoadingSpinner() {
+
+        loadingSpinner.showFor("refresh-employee-roster");
+
+        return refreshWithoutLoadingSpinner().then(i -> {
+            loadingSpinner.hideFor("refresh-employee-roster");
+            return resolve();
+        }).catch_(e -> {
+            ErrorPopup.show(e.toString());
+            loadingSpinner.hideFor("refresh-employee-roster");
+            return resolve();
+        });
+    }
+
+    //Events
+    @EventHandler("refresh-button")
+    public void onRefreshButtonClicked(@ForEvent("click") final MouseEvent e) {
+        refreshWithLoadingSpinner();
+    }
+
+    @EventHandler("previous-page-button")
+    public void onPreviousPageButtonClicked(@ForEvent("click") final MouseEvent e) {
+
+        /*if (spotsPagination.isOnFirstPage()) {
+            return;
+        }
+        
+        spotsPagination = spotsPagination.previousPage();
+        refreshWithLoadingSpinner();*/
+    }
+
+    @EventHandler("next-page-button")
+    public void onNextPageButtonClicked(@ForEvent("click") final MouseEvent e) {
+        /*spotsPagination = spotsPagination.nextPage();
+        refreshWithLoadingSpinner();*/
+    }
+
+    //FIXME: Improve horizontal navigation. Probably snap to fixed dates with animation.
+    private static final Integer TIME_SCROLL_SIZE = 300;
+
+    @EventHandler("forward-in-time-button")
+    public void onForwardInTimeButtonClicked(@ForEvent("click") final MouseEvent e) {
+        viewportFrame.scrollLeft += TIME_SCROLL_SIZE;
+    }
+
+    @EventHandler("back-in-time-button")
+    public void onBackInTimeButtonClicked(@ForEvent("click") final MouseEvent e) {
+        viewportFrame.scrollLeft -= TIME_SCROLL_SIZE;
+    }
+
+    //API calls
```

Comment:
I'd remove this comment - Seems obvious from the use of "..RestService..."

Reply:
It to separate method types, to make it easier to find what method does a particular thing